src/raw.rs

```diff
@@ -388,7 +386,7 @@ impl Future for Connect {
 pub struct Client;
 
 impl Client {
-    #![cfg_attr(feature = "cargo-clippy", allow(new_ret_no_self))]
+    #![cfg_attr(feature = "cargo-clippy", allow(clippy::new_ret_no_self))]
```

Comment:
This can be changed to `#![allow(clippy::new_ret_no_self)]` now. So as the transaction client.